api/v1alpha1/virtualmachine_types.go

```diff
@@ -9,14 +9,17 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-type VirtualMachinePowerState string
-
 // See govmomi.vim25.types.VirtualMachinePowerState
 const (
 	VirtualMachinePoweredOff VirtualMachinePowerState = "poweredOff"
 	VirtualMachinePoweredOn  VirtualMachinePowerState = "poweredOn"
 )
 
+// VirtualMachinePowerState represents the power state of a VirtualMachine.
+// The value values are "poweredOn", and "poweredOff".
+// +kubebuilder:validation:Enum=poweredOff;poweredOn
```

Comment:
I think today we have situations where the status field will be the default value (""), and it isn't clear to me that this would allow that.

Reply:
I tried out the service with this change. As of now, we do not update the power state with "" anywhere. We either don't fill it out in the `VirtualMachineStatus`, or fill valid values (`poweredOff`, `poweredOn`). As long as we continue to do that, we won't have any issues.

Reply:
I am marking this as resolved.